src/blockchain/Blockchain.ts

```diff
@@ -141,28 +208,135 @@ export default class Blockchain implements IBlockchainApi {
     return this.api.rpc.chain.subscribeNewHeads(listener)
   }
 
+  /**
+   * [ASYNC] Signs the SubmittableExtrinsic with the given identity.
+   *
+   * @param identity The [[Identity]] to sign the Tx with.
+   * @param tx The unsigned SubmittableExtrinsic.
+   * @returns Signed SubmittableExtrinsic.
+   *
+   */
   public async signTx(
     identity: Identity,
     tx: SubmittableExtrinsic
   ): Promise<SubmittableExtrinsic> {
     const nonce = await this.getNonce(identity.address)
-    const signed: SubmittableExtrinsic = identity.signSubmittableExtrinsic(
+    const signed: SubmittableExtrinsic = await identity.signSubmittableExtrinsic(
       tx,
-      nonce.toHex()
+      nonce
     )
     return signed
   }
 
+  /**
+   * [ASYNC] Submits a signed [[SubmittableExtrinsic]] with exported function [[submitSignedTx]].
+   * Handles recoverable errors if identity is provided by re-signing and re-sending the tx up to two times.
+   * Uses parseSubscriptionPromise to provide complete potentially defaulted options to the called submitSignedTx.
+   *
+   * Transaction fees will apply whenever a transaction fee makes it into a block, even if extrinsics fail to execute correctly!
+   *
+   * @param tx The [[SubmittableExtrinsic]] to be submitted. Most transactions need to be signed, this must be done beforehand.
+   * @param identity Optional [[Identity]] to potentially re-sign the Tx with.
+   * @param opts Optional partial criteria for resolving/rejecting the promise.
+   * @returns A promise which can be used to track transaction status.
+   * If resolved, this promise returns the eventually resolved [[SubmittableResult]].
+   */
+  public async submitSignedTx(
```

Comment:
Same name as the function export is confusing. What does this method do differently?

Reply:
`submitTxWithResign`?

Reply:
I like `submitTxWithResign`, but we have many similarly named functions that makes this class kinda convoluted.
After refactoring of the tx signing and submission procedure, this should be more on point.